### ISSUE TITLE
Error::Read when attempting to read from an NVR API

### ISSUE THREAD

**Issue by ASBaj** [yhirose/cpp-httplib#1172]
Hi,

I'm building an application for pulling data from an API of a NVR we use. During development, I encountered no issues in pulling the data I needed. However, as I am starting to move the code into production, I have found that there are a couple of NVRs that cause the library to throw a Error::Read. I attempted to step through the code as far as possible, but was only able to find that the strm_.read() function during stream_line_reader::getline() was returning 0, and the i variable in the loop was also 0, so the function would therefore return false. I am not sure why this is returning the way this.

I have not been able to replicate this issue via the browser, curl, or Postman, and am therefore confused as to what is going on here. What is odd about this particular issue is that only this type of NVR seems to have the problem; yet if it the NVR was the source of the issue, I would think that I should be able to replicate the problem outside of this library. Do you have any ideas for assisting me in determining if this is a library issue?

**Comment by ASBaj** [comment]
Update: I found that in read_headers, line_reader.end_with_crlf() is failing, meaning no headers are parsed. I'm not entirely sure why the check is failing though.

**Comment by ASBaj** [comment]
Another update: the buffer used to populate the line reader contains the following "Status: 401\n". When testing it against a working NVR, the working NVR returns "Status: 401\r\n". It would appear the NVR is not sending back CRLF, only LF, based on my quick google searches about the subject. This indicates less of a library problem and more of an NVR one. This does lead to the next question though: can I bypass this requirement for httplib to work?

**Comment by yhirose** [comment]
@ASBaj, thank you for the report. I found the following articles explaining the situation that you discoverted.

https://stackoverflow.com/questions/5757290/http-header-line-break-style
https://stackoverflow.com/questions/43574428/have-apache-accept-lf-vs-crlf-in-request-headers

I am ok to introduce a macro symbol like `CPPHTTPLIB_UNSAFE` to accept such an incorrect format as Apache does. But, is it better to ask the developer of the NVR API to correct the line end problem first?

**Comment by ASBaj** [comment]
@yhirose Thanks for the reply, and adding in some kind of workaround for this situation is greatly appreciated.

The developer of the NVR has actually fixed this is newer models. However, I've got quite a few older models that I need to support in the long term. One of the first things I checked was if I could update the firmware of the NVR, and unfortunately, there are no updates available for these models.

Hopefully it isn't too much trouble to resolve this issue. Thanks!
